core/src/main/java/io/dekorate/Session.java

```diff
@@ -166,19 +166,19 @@ public void addConfiguration(Map<String, Object> map, BiConsumer<ConfigurationGe
       String key = entry.getKey();
       Object value = entry.getValue();
       ConfigurationGenerator generator = configurationGenerators.get(key);
-      if (generator == null) {
-        throw new IllegalArgumentException(
-            "Unknown generator '" + key + "'. Known generators are: " + configurationGenerators.keySet());
-      }
-
-      if (value instanceof Map) {
-        Map<String, Object> generatorMap = new HashMap<>();
-        Class configClass = configtypes.get(key);
-        String newKey = configClass.getName();
-        Generators.applyPrimitives(configClass, (Map<String, Object>) value);
-        Generators.populateArrays(configClass, (Map<String, Object>) value);
-        generatorMap.put(newKey, value);
-        consumer.accept(generator, Maps.kebabToCamelCase(generatorMap));
+      if (generator != null) {
+        if (value instanceof Map) {
```

Comment:
I had to make this change because, during the build, we're adding some `dekorate.docker.xxx` system properties. As part of this PR, we're now including the system properties, therefore some examples were failing because the `docker` generator didn't exist (the `docker-annotations` module was not in the Maven configuration).

Having said that, I do think the right behavior is to not fail, but simply add a warning in the logs to let users know. 
What do you think about this change? @iocanel

Reply:
I think that this brings to the surface that problems that arise from not having profile support or something like this.
So, I'd say it's fair to relax this check.